### PR TITLE
#53 Reproduce behaviour with generic methods

### PR DESCRIPTION
**Pull request by sergey-karazhenets**

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this pull request

**Comment by 0crat** [comment]
Job `gh:yegor256/jpeek#54` is in scope.

File: src/test/java/org/jpeek/metrics/cohesion/CAMCTest.java [hunk 1]
```diff
@@ -39,6 +39,8 @@
  * @since 0.1
  * @checkstyle AbbreviationAsWordInNameCheck (5 lines)
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle EmptyLinesCheck (500 lines)
+ * @checkstyle LineLengthCheck (500 lines)
```
Comment (yegor256) [inline src/test/java/org/jpeek/metrics/cohesion/CAMCTest.java]: @sergey-karazhenets please, get rid of them both. It's a bad practice to have empty lines in methods: http://www.yegor256.com/2014/11/03/empty-line-code-smell.html Also, when you _need_ to have a long line, put the Checkstyle suppression right in front of it

**Comment by yegor256** [comment]
@sergey-karazhenets thanks! See above

**Comment by yegor256** [comment]
@sergey-karazhenets what's up with this one?

**Comment by 0crat** [comment]
Job `gh:yegor256/jpeek#54` assigned to [@icok](http://www.0crat.com/u/icok}) (`DEV`). The budget is [fixed](http://datum.zerocracy.com/pages/policy.html#4) and it is 30 minutes. Please, read [§4](http://datum.zerocracy.com/pages/policy.html#4), [§8](http://datum.zerocracy.com/pages/policy.html#8), and [§9](http://datum.zerocracy.com/pages/policy.html#9). If the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html).

**Comment by iCok** [comment]
@0crat waiting

**Comment by iCok** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/jpeek/issues/54#issuecomment-356349422))

@iCok The impediment was registered successfully

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/jpeek/issues/54#issuecomment-356857525))

@iCok I can't do it for technical reasons, I'm very sorry.  If you don't know what to do,  submit this error as a ticket  [here](https://github.com/zerocracy/datum):

```
 1.0-SNAPSHOT cf14496c 2018-01-10T18:23:00
IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pm/in/impediments/register_impediment.groovy
java.lang.IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pm/in/impediments/register_impediment.groovy
	at com.zerocracy.farm.reactive.StkGroovy.process(StkGroovy.java:125)
	at com.zerocracy.farm.StkSafe.process(StkSafe.java:88)
	at com.zerocracy.farm.reactive.StkSmart.process(StkSmart.java:66)
	at com.zerocracy.farm.reactive.Brigade.process(Brigade.java:82)
	at com.zerocracy.farm.reactive.Brigade.lambda$apply$0(Brigade.java:64)
	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:58)
	at org.cactoos.func.IoCheckedBiFunc.apply(IoCheckedBiFunc.java:66)
	at org.cactoos.func.UncheckedBiFunc.apply(UncheckedBiFunc.java:60)
	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:59)
	at org.cactoos.iterator.Filtered.hasNext(Filtered.java:90)
	at org.cactoos.iterator.LengthOf.lambda$new$0(LengthOf.java:52)
	at org.cactoos.scalar.UncheckedScalar.lambda$value$0(UncheckedScalar.java:58)
	at org.cactoos.func.UncheckedFunc.lambda$apply... 
```

Cc [@yegor256](http://www.0crat.com/u/yegor256)

**Comment by iCok** [comment]
@yegor256 should I review or fix some issue here? Could you provide more information related to the task?

**Comment by 0crat** [comment]
[@icok](http://www.0crat.com/u/icok) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by iCok** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/yegor256/jpeek/issues/54#issuecomment-357504832))

@iCok Job [#54](https://github.com/yegor256/jpeek/issues/54) is already on hold

**Comment by iCok** [comment]
@yegor256 0crat notifies me about deadline of that task, but requirements still are not clear, can you provide more information?

**Comment by yegor256** [comment]
@eugene-nikolaev this is not valid anymore, since our mechanism of metrics calculation is totally different now. Sorry.

**Comment by eugene-nikolaev** [comment]
I guess that was actually for @sergey-karazhenets

**Comment by 0crat** [comment]
The job [#54](https://github.com/yegor256/jpeek/issues/54) is now out of scope

**Comment by 0crat** [comment]
Order was successfully finished: +30 points just awarded to [@icok](http://www.0crat.com/u/icok), total is [+30](http://www.0crat.com/u/icok)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to [@yegor256](http://www.0crat.com/u/yegor256), total is [+7255](http://www.0crat.com/u/yegor256)

**Pull request closed without merging**